WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorMedia.kt

```diff
@@ -157,37 +150,37 @@ class EditorMedia(
     }
 
     fun advertiseImageOptimisationAndAddMedia(data: Intent) {
-        if (WPMediaUtils.shouldAdvertiseImageOptimization(activity)) {
-            WPMediaUtils.advertiseImageOptimization(activity) { addMediaItemGroupOrSingleItem(data) }
+        if (mediaUtilsWrapper.shouldAdvertiseImageOptimization()) {
+            mediaUtilsWrapper.advertiseImageOptimization { addMediaItemGroupOrSingleItem(data) }
         } else {
             addMediaItemGroupOrSingleItem(data)
         }
     }
 
     fun addExistingMediaToEditor(source: AddExistingMediaSource, mediaId: Long): Boolean {
-        val media = mediaStore.getSiteMediaWithId(site, mediaId)
-        if (media == null) {
-            AppLog.w(T.MEDIA, "Cannot add null media to post")
-            return false
+        mediaStore.getSiteMediaWithId(site, mediaId)?.let { media ->
```

Comment:
We may want to consider improving this so the method has only one return statement. Imho it's a good practice + a very subtle change to the code will break it and it'll be extremely difficult to catch the issue. 

Perhaps something like this:
```
fun addExistingMediaToEditor(source: AddExistingMediaSource, mediaId: Long): Boolean {
        return mediaStore.getSiteMediaWithId(site, mediaId)?.let { media ->
            editorTracker.trackAddMediaEvent(site, source, media)
            fluxCUtilsWrapper.mediaFileFromMediaModel(media)?.let { mediaFile ->
                editorMediaListener.appendMediaFile(mediaFile, media.urlToUse)
            }
            true
        } ?: false.also { AppLog.w(T.MEDIA, "Cannot add null media to post") }
    }
```
Wdyt?


SideNote:
Just so the comment make sense. Notice that if I remove the "return" the code won't work, however, noticing the bug will be very difficult.

```
fun addExistingMediaToEditor(source: AddExistingMediaSource, mediaId: Long): Boolean {
        mediaStore.getSiteMediaWithId(site, mediaId)?.let { media ->
            editorTracker.trackAddMediaEvent(site, source, media)
            fluxCUtilsWrapper.mediaFileFromMediaModel(media)?.let { mediaFile ->
                editorMediaListener.appendMediaFile(mediaFile, media.urlToUse)
            }
           true <-------- This will return "true" from the let, but not from the `addExistingMediaToEditor`
        }
        AppLog.w(T.MEDIA, "Cannot add null media to post")
        return false
    }
```

Reply:
Fixed in 5b70d2423f951be95ce43313e7d6b40c1c53c8ac.